### DISCUSSION TITLE
[mini.pairs] Python docstring

### DISCUSSION THREAD

**Discussion by dhruvinsh** [nvim-mini/mini.nvim#255] [Q&A]
Hi, 
Python docstring, usually surround by 6 quote as below,
``` python
def foo():
    """
    This is a python doc string
    """
```
Is it possible to acheive with mini.pairs where inserting `"""` makes `""" """`?

**Comment by echasnovski** [comment]
Sorry, no. By design, 'mini.pairs' works only on with two-character pairs (like `()`, `{}`, etc.) and can be conditioned only by two-character neighborhood (with `neigh_pattern`). This functionality is more suited for snippets.

**Reply by dhruvinsh** [reply]
Thank you so much no need to apologies 😃 . You have contributed enough.

Just for reference, any plan expanding this functionality more then two charecters?

**Reply by echasnovski** [reply]
> Just for reference, any plan expanding this functionality more then two charecters?

No, not in 'mini.pairs'. The 'mini.snippets' module is planned, though.

**Comment by YoungStovallbL** [comment]
Hey there!

So it looks like you're wondering if it's possible to use the mini.pairs function in Python to surround a docstring with six quotes instead of the usual three, is that right?

Unfortunately, it sounds like that's not possible with mini.pairs. The function is designed to work with two-character pairs (like parentheses or curly braces) and can only be conditioned by a two-character neighborhood. It's really more suited for use with code snippets.

However, the good news is that the mini.snippets module is planned for future development, which should provide expanded functionality for this type of thing.

**Comment by Susensio** [comment]
Just for reference, I was very annoyed that typing 3 `"` inserted the fourth unwanted quote
```
"""|"
```

This, at least, solves that annoyance
```
mappings = {
  ['"'] = { action = 'closeopen', pair = '""', neigh_pattern = '[^\\"].', register = { cr = false } },
}
```
It uses prev neigh pattern to disable opening if right of quote. ~The downside is that you cannot insert quotes inside quotes but.. who does that?~ That wasn't possible bc closing action would skip the next quote, so... I like this solution even more
